packages/components/scripts/vendorism/custom/input.vendorism.js

```diff
@@ -1,42 +1,252 @@
-  const output = { content, path };
-  if (!path.includes('input.component.ts')
-    && !path.includes('input.styles.ts')
-    && !path.includes('input.stories.ts')
-    && !path.includes('input.test.ts')
-  ) {
-    return output;
-  }
-  // We don't provide a filled property in Synergy, but instead use the filled styles for readonly
-  // update stories
-  output.content = output.content.replace('<syn-input placeholder="Type something" filled>', '<syn-input value="Readonly content" readonly>');
-  output.content = output.content.replaceAll('Filled', 'Readonly');
-
-  // update component and styles
-  output.content = removeSection(output.content, '/** Draws a filled', 'filled = false;');
-  output.content = output.content.replaceAll('filled', 'readonly'); // makes changes in styles and components
-  // Pill is not supported in Synergy
-  output.content = removeSection(output.content, "'input--pill':", ',');
-  output.content = removeSection(output.content, '/** Draws a pill-style', 'pill = false;');
-  // We need to add classes depenending on prefix and suffix slots to use them in CSS
-  output.content = output.content.replace(
+const FILES_TO_TRANSFORM = [
+  'input.component.ts',
+  'input.styles.ts',
+  'input.test.ts',
+];
+
+/**
+ * Transform the component code
+ * @param {String} path
+ * @param {String} originalContent
+ * @returns
+ */
+const transformComponent = (path, originalContent) => {
+  let content = removeSections([
+    ['/** Draws a filled', 'filled = false;'],
+    ['/** Draws a pill-style', 'pill = false;'],
+    ["'input--pill':", ','],
+  ], originalContent);
+
+  // Replace filled with readonly
+  content = content.replaceAll('filled', 'readonly');
+
+  // We need to add classes depending on prefix and suffix slots to use them in CSS
+  content = content.replace(
     "HasSlotController(this, 'help-text', 'label')",
     "HasSlotController(this, 'help-text', 'label', 'prefix', 'suffix')",
   );
-  output.content = output.content.replace(
+  content = content.replace(
     "const hasHelpTextSlot = this.hasSlotController.test('help-text');",
     "const hasHelpTextSlot = this.hasSlotController.test('help-text');\n    const hasPrefixSlot = this.hasSlotController.test('prefix');\n    const hasSuffixSlot = this.hasSlotController.test('suffix');",
   );
-  output.content = output.content.replace(
+  content = content.replace(
     "'form-control--has-help-text': hasHelpText",
     "'form-control--has-help-text': hasHelpText,\n          'form-control--has-prefix': hasPrefixSlot,\n          'form-control--has-suffix': hasSuffixSlot",
   );
-  // remove tests for pill and filled
-  output.content = removeSection(output.content, 'expect(el.filled)', ';');
-  output.content = removeSection(output.content, 'expect(el.pill)', ';');
-  // @todo remove this when syn-textarea and syn-checkbox are available
-  output.content = output.content.replace(/syn-checkbox/g, 'syn-input');
+
+  // Replace browser number built-in comments
```

Comment:
Finger's crossed, that this will survive the next Shoelace updates. :D

Reply:
Which are by the way there, too. Maybe we should check against the latest version to make sure?

Reply:
Tested out the new shoelace version 2.13.1. There were no changes on the input, which will lead to problems. But we need to have a special look at it when updating, because the build does no longer run through (creating wrappers, custom-elements.json looks different now, etc). I will create a ticket for updating shoelace